### PR TITLE
Adding save file functionality to query plan

### LINKED ISSUE #18289: Add ability to save query plan as  .sqlplan file
Add action to context menu to save query plan as .sqlplan file

image

### PR DESCRIPTION
**Pull request by aasimkhan30**
This PR Fixes #18289
image
image

### CODE REVIEW

File: src/sql/workbench/contrib/queryplan2/browser/queryPlan.ts [hunk 1]
```diff
@@ -515,6 +520,16 @@ class SavePlanFile extends Action {
 	}
 
 	public override async run(context: QueryPlan2): Promise<void> {
+		const saveFileUri = await context.fileDialogService.showSaveDialog({
```
Comment (alanrenmsft) [inline src/sql/workbench/contrib/queryplan2/browser/queryPlan.ts:523]: 1. you should give a default file name
2. need to handle the save dialog canceled scenario.

File: src/sql/workbench/contrib/queryplan2/browser/queryPlan.ts [hunk 2]
```diff
@@ -515,6 +522,23 @@ class SavePlanFile extends Action {
...
+		const workspaceFolders = await context.workspaceContextService.getWorkspace().folders;
+		let currentWorkSpaceFolder;
+		if (workspaceFolders.length !== 0) {
+			currentWorkSpaceFolder = workspaceFolders[0].uri;
+		}
+		const saveFileUri = await context.fileDialogService.showSaveDialog({
+			filters: [
+				{
+					extensions: ['sqlplan'], //TODO: Get this extension from provider
+					name: localize('queryPlan.SaveFileDescription', 'Execution Plan Files')
```
Comment (alanrenmsft) [inline src/sql/workbench/contrib/queryplan2/browser/queryPlan.ts:534]: this should also come from extension in the future, right?

**Review by alanrenmsft — APPROVED** [review]
please give a proper default file name, otherwise looks good to me.

File: src/sql/workbench/contrib/queryplan2/browser/queryPlan.ts (hunk 2 above)
Comment (aasimkhan30) [reply src/sql/workbench/contrib/queryplan2/browser/queryPlan.ts:534]: Yes, both the extension and name should come from provider

**Pull request merged**